### DISCUSSION TITLE
What does this `extract_standalone` do?

### DISCUSSION THREAD

**Discussion by runnerSnail** [denoland/deno#9576] [Q&A]
# Why do you need to call 'standalone::extract_standalone' before using 'run xb.ts'? What does this function do
## question 1
![image](https://user-images.githubusercontent.com/40654631/108679595-ba050e80-7527-11eb-894c-63d7344b2603.png)
## question2 
> pub const MAGIC_TRAILER: &[u8; 8] = b"d3n0l4nd"; // what?

**Comment by lucacasonato** [comment]
This is used for `deno compile` binaries. It works by appending a bundled script to the end of a regular (or "lite" binary), and then executing it as usual. This `extract_standalone` makes it work. See https://github.com/denoland/deno/pull/8539
